File: examples/rust/add-raw-output/RawOutputTest_test.rs
// RawOutputTest.runar.rs uses `self.add_raw_output` and `self.add_output`
// — Rúnar intrinsics that the compiler materialises into the emitted
// Bitcoin Script but that do not exist as methods on the generated Rust
// struct. Attempting `#[path] mod contract;` therefore fails to compile
// (the methods are not on &mut RawOutputTest).
//
// NFT_test.rs solves the same problem by defining a shadow struct inline;
// for this much smaller contract we simply exercise the Rúnar frontend
// directly via compile_check, which is the cross-compiler conformance
// boundary we care about.

#[test]
fn test_compile() {
    runar::compile_check(
        include_str!("RawOutputTest.runar.rs"),
        "RawOutputTest.runar.rs",
    )
    .unwrap();
}

File: examples/go/add-raw-output/RawOutputTest_test.go
package contract

import (
	"testing"

	runar "github.com/icellan/runar/packages/runar-go"
)

// RawOutputTest.runar.go uses self.AddRawOutput and self.AddOutput — Rúnar
// intrinsics the compiler materialises into emitted Bitcoin Script. The
// cross-compiler conformance boundary we care about is the Rúnar frontend
// (parse → validate → typecheck), so we exercise that directly via
// CompileCheck.

func TestRawOutputTest_Compile(t *testing.T) {
	if err := runar.CompileCheck("RawOutputTest.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}
